### PR TITLE
revert #449

### PR DESCRIPTION
**Pull request by oscardssmith**
Since the SArray types are created in StaticArraysCore, unfortunately we need the extension to be on the library or else generated functions in 3rd party packages can miss the added methods. Furthermore, this breaks the ability to optimize `lu` since `StaticArraysCore` doesn't define `StaticArrays.LU` so this just is really annoying. I really think we should delete StaticArraysCore and just make it an alias for StaticArrays now that extension packages exist.

### CODE REVIEW

File: ext/ArrayInterfaceStaticArraysCoreExt.jl [hunk 1]
```diff
@@ -21,10 +21,6 @@ ArrayInterface.can_setindex(::Type{<:StaticArray}) = false
 ArrayInterface.can_setindex(::Type{<:MArray}) = true
 ArrayInterface.buffer(A::Union{SArray, MArray}) = getfield(A, :data)
 
-function ArrayInterface.lu_instance(A::SMatrix{N,N}) where {N}
```
Comment (ChrisRackauckas) [inline ext/ArrayInterfaceStaticArraysCoreExt.jl:24]: why was this removed?

Comment (oscardssmith) [reply ext/ArrayInterfaceStaticArraysCoreExt.jl:24]: because `LU` is defined in `StaticArrays` not `StaticArraysCore`

Comment (ChrisRackauckas) [reply ext/ArrayInterfaceStaticArraysCoreExt.jl:24]: Didn't we need this for performance? What about having a StaticArraysExt for just this one?

Comment (oscardssmith) [reply ext/ArrayInterfaceStaticArraysCoreExt.jl:24]: I tried that locally and it causes invalidations during precompile which is illegal. I'll just hack my way around it...
